### PR TITLE
Scottx611x/ensure workflow outputs exist

### PR DESCRIPTION
**Pull request by scottx611x**
- Ensure that a user has properly set `workflow_outputs` for their Galaxy Workflow before allowing the creation of a ToolDefinition.
- Also Handle one-off error exhibited [here](https://gist.github.com/scottx611x/3dd5e79d405547454bc0abbf233017e3)

Towards #2294

### CODE REVIEW

**Comment by scottx611x** [comment]
@hackdna @mccalluc Could you check this out when you get a chance?

**Comment by hackdna** [comment]
Just to clarify: this will be rewritten when we go back to step annotations in 1.6.2, correct?

**Comment by scottx611x** [comment]
@hackdna Yes

File: refinery/tool_manager/management/commands/generate_tool_definitions.py [hunk 1]
```diff
@@ -178,6 +184,47 @@ def _generate_workflows(self):
                     )
                 )
 
+    def ensure_workflow_outputs_are_present(self, workflow):
```
[the remark this answers is not in the document]
Comment (mccalluc) [reply refinery/tool_manager/management/commands/generate_tool_definitions.py]: Ditto: From the name, someone might think it adds the workflow outputs if missing.

naming: even more concise: `has_workflow_outputs()`. (but `has_outputs()` would be too short?)

File: refinery/tool_manager/management/commands/generate_tool_definitions.py [hunk 2]
```diff
@@ -178,6 +182,47 @@ def _generate_workflows(self):
...
+        """
+        Assert that the given `workflow` has `workflow_outputs` properly
+        defined
+        :param workflow: dict containing a Galaxy Workflow's information
+        :raises: CommandError
+        """
+        workflow_steps = workflow["graph"]["steps"]
+        for step_index in workflow_steps:
+            step_has_workflow_outputs = workflow_steps[step_index].get(
```
Comment (mccalluc) [inline refinery/tool_manager/management/commands/generate_tool_definitions.py]: This isn't *really* a boolean; probably no harm done, but still.

File: refinery/tool_manager/management/commands/generate_tool_definitions.py [hunk 3]
```diff
@@ -178,6 +182,47 @@ def _generate_workflows(self):
...
+                WorkflowTool.WORKFLOW_OUTPUTS
+            )
+            if step_index == "0":
+                if not step_has_workflow_outputs:
+                    raise CommandError(
+                        "You've encountered known error when trying to "
```
Comment (mccalluc) [inline refinery/tool_manager/management/commands/generate_tool_definitions.py]: "*a* known error"

**Pull request merged**